spec/rails_helper.rb

```diff
@@ -50,6 +50,9 @@
 
   config.include FactoryBot::Syntax::Methods
 
+  config.before(:suite) do
+    ActiveJob::Base.queue_adapter = :test
```

Comment:
If Redis has been configured in the test environment, and Redis is only present to support Sidekiq, then I suspect that we should be testing against Sidekiq here. https://github.com/philostler/rspec-sidekiq ?

Reply:
We're not actually using sidekiq in a way that allows us to test using the rspec-sidekiq gem.
As far as I can tell, it's intended for when sidekiq is used via 'include Sidekiq::Worker' in the worker classes.
Since we use ActiveJob and Sidekiq only as an adapter it makes sense to just test with the default configuration for testing ActiveJob jobs, which is to use the :test adapter in the test environment.
I've removed the redis url from the redis config since we don't need to have redis running for tests at all.
In fact we don't need to run the redis container in the test environment, do you have any idea how we could detect that we're running in a test only evnironment for the likes of travis?
Maybe setting a flag when we call 'bundle exec ./docker/setup' in the before_script section of our .travis.yml, and then picking that up to skip the redis container?

Reply:
If rspec-sidekiq is not of use to us, that is fine. But we should be running Redis & Sidekiq in any test environments including Travis. If something is broken in our stack, we need to know about it before it gets out to production, hence the need for homogeneity across environments/deployments.

A feature spec would run one of these jobs through Sidekiq and then check for the presence of the thumbnails in the S3 storage service.